sqlmesh/utils/date.py

```diff
@@ -295,33 +297,40 @@ def make_inclusive(start: TimeLike, end: TimeLike) -> DatetimeRange:
     In the ds ('2020-01-01') case, because start_ds and end_ds are categorical, between works even if
     start_ds and end_ds are equivalent. However, when we move to ts ('2022-01-01 12:00:00'), because timestamps
     are numeric, using simple equality doesn't make sense. When the end is not a categorical date, then it is
-    treated as an exclusive range and converted to inclusive by subtracting 1 microsecond.
+    treated as an exclusive range and converted to inclusive by subtracting 1 nanosecond.
 
     Args:
         start: Start timelike object.
         end: End timelike object.
 
     Example:
         >>> make_inclusive("2020-01-01", "2020-01-01")
-        (datetime.datetime(2020, 1, 1, 0, 0, tzinfo=datetime.timezone.utc), datetime.datetime(2020, 1, 1, 23, 59, 59, 999999, tzinfo=datetime.timezone.utc))
+        (datetime.datetime(2020, 1, 1, 0, 0, tzinfo=datetime.timezone.utc), Timestamp('2020-01-01 23:59:59.999999999+0000', tz='UTC'))
 
     Returns:
         A tuple of inclusive datetime objects.
     """
+
     return (to_datetime(start), make_inclusive_end(end))
 
 
 def make_inclusive_end(end: TimeLike) -> datetime:
-    return make_exclusive(end) - timedelta(microseconds=1)
+    return make_exclusive(end) - pd.Timedelta(1, unit="ns")
 
 
 def make_exclusive(time: TimeLike) -> datetime:
```

Comment:
is this type still right is datetime and pd.Timedelta a subclass?

Reply:
yep, for example:

```python
>>> pd.Timestamp("2020-01-01") + pd.Timedelta(1, unit="day")
Timestamp('2020-01-02 00:00:00')
>>> issubclass(pd.Timestamp, datetime.datetime)
True
```